Prose/ProseLib/Sources/App/AppReducer.swift

```diff
@@ -35,7 +35,7 @@ public let appReducer: Reducer<
                 sidebar: .init(credentials: .init(jid: jid))
             ))
 
-        case let .auth(.loginResult(.success(jid))):
+        case let .auth(.didLogIn(jid, _)):
```

Comment:
I was just thinking if there could be a login failure and who should handle that. Of course the `authenticationReducer` for all but one case: When the user cancels. The app should quit if you close the authentication window*. Currently you're just ending up with a windowless application if you close it. How can we implement this in SwiftUI?

*We're going to reuse the `AuthenticationFeature` if you log into other accounts. If you're already logged into at least one, the app should not quit of course. This is where it starts to make sense that the parent - in this case the `appReducer` decides how to handle the result.

We could have a struct in our SharedModels: `UserCancelledError`. The enum case would then be `.loginResult(Result<Credentials, UserCancelledError>)`.

Reply:
Do I keep this as we got rid of the secondary window?

Reply:
Keep what exactly? The login still needs the functionality to be cancelled. We could just say that if there is no account, we don't show a close/cancel button. If we do, the app should quit when it's tapped. If there is an account the sheet should be dismissed.

Reply:
I meant "keep this as is". But you're right, we need such feature. Can we just keep it for after #38?